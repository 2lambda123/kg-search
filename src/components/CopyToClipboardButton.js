/*
*   Copyright (c) 2018, EPFL/Human Brain Project PCO
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*/

import React from "react";
import PropTypes from "prop-types";
import "./CopyToClipboardButton.css";

export const CopyToClipboardButton = props => {

  const clickHandler = (event) => {

    var textArea = document.createElement("textarea");
    textArea.style.position = "absolute";
    textArea.style.top = 0;
    textArea.style.left = 0;
    textArea.style.width = "2em";
    textArea.style.height = "2em";
    textArea.style.padding = 0;
    textArea.style.margin = 0;
    textArea.style.border = "none";
    textArea.style.outline = "none";
    textArea.style.boxShadow = "none";
    textArea.style.background = "transparent";
    textArea.style.color = "transparent";

    textArea.value = props.content;

    const button = event.currentTarget;

    button.appendChild(textArea);

    textArea.select();

    try {
      document.execCommand("copy");

      button.setAttribute("show", "true");
      setTimeout(() => button.removeAttribute("show"), 1000);

    } catch (e) {
      //window.console.debug("could not run execCommand copy");
    }

    button.removeChild(textArea);
  };

  let title = "Send to clipboard";
  if (props.title) {
    title = props.title;
  }

  let confirmationText = "sent to clipoard";
  if (props.confirmationText) {
    confirmationText = props.confirmationText;
  }

  let iconClassName = "fa fa-clipboard";
  if (props.icon) {
    iconClassName = props.icon;
  }

  let icon = null;
  let text = null;
  if (props.text) {
    if (props.icon) {
      icon = <i className={iconClassName}></i>;
    }
    text = <span>{props.text}</span>;
  } else {
    icon = <i className={iconClassName}></i>;
  }

  const classNames = ["kgs-copy-to-clipboard", props.className].join(" ");

  return (
    <span className={classNames}>
      <button role="link" onClick={clickHandler} title={title}>{icon}{text}</button>
      <div className="kgs-copy-confirmation">{confirmationText}</div>
    </span>
  );
};

CopyToClipboardButton.propTypes = {
  content: PropTypes.string,
  title: PropTypes.string,
  confirmationText: PropTypes.string,
  icon: PropTypes.string
};

export default CopyToClipboardButton;