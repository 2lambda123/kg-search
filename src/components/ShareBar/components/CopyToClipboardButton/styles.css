/*
*   Copyright (c) 2018, EPFL/Human Brain Project PCO
*
*   Licensed under the Apache License, Version 2.0 (the "License");
*   you may not use this file except in compliance with the License.
*   You may obtain a copy of the License at
*
*       http://www.apache.org/licenses/LICENSE-2.0
*
*   Unless required by applicable law or agreed to in writing, software
*   distributed under the License is distributed on an "AS IS" BASIS,
*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
*   See the License for the specific language governing permissions and
*   limitations under the License.
*/

.kgs-copy-to-clipboard {
    display: inline-block;
}

.kgs-copy-to-clipboard button {
    display: inline-block;
    margin: 0;
    padding: 0 10px;
    border: 0;
    border-radius: 3px;
    background-color: transparent;
    font-weight: 500;
    font-size: 14px;
    color: rgb(108, 115, 121);
    font-size: 22px;
    text-align: center;
    -webkit-transition: color 0.2s ease-in, -webkit-box-shadow 0.2s ease-in;
    transition: box-shadow 0.2s ease-in, color 0.2s ease-in, -webkit-box-shadow 0.2s ease-in;
}

.kgs-copy-to-clipboard:hover {
    color: rgb(51, 51, 51);
    -webkit-box-shadow: 3px 3px 6px #cbc9c9;
    box-shadow: 3px 3px 6px #cbc9c9;
}

.kgs-copy-confirmation {
    display: none;
    position: absolute;
    width: 170px;
    margin-top: -100px;
    margin-left: -56px;
    padding: 4px 8px;
    border-radius: 20px;
    font-size: 10px;
    text-align: center;
    color: transparent;
    background-color: transparent;
    transition: color 0.25s cubic-bezier(.34,1.06,.63,.93), background-color 0.25s cubic-bezier(.34,1.06,.63,.93),margin-top 0.5s cubic-bezier(.34,1.06,.63,.93);
}

[data-hashits] .kgs-copy-confirmation,
[data-showdetail="true"] .kgs-copy-confirmation {
    display: block;
}

[show="true"] + .kgs-copy-confirmation{
    margin-top: 5px;
    color: #333;
    background-color: #ffeeaa;
}
