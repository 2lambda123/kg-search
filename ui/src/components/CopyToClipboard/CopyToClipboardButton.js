/*
 * Copyright 2018 - 2021 Swiss Federal Institute of Technology Lausanne (EPFL)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This open source software code was developed in part or in whole in the
 * Human Brain Project, funded from the European Union's Horizon 2020
 * Framework Programme for Research and Innovation under
 * Specific Grant Agreements No. 720270, No. 785907, and No. 945539
 * (Human Brain Project SGA1, SGA2 and SGA3).
 *
 */

import React from "react";
import PropTypes from "prop-types";
import "./CopyToClipboardButton.css";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

export const CopyToClipboardButton = props => {

  const clickHandler = (event) => {

    const div = document.createElement("div");
    div.innerHTML = props.content;

    const content = div.innerText;

    const textArea = document.createElement("textarea");
    textArea.style.position = "absolute";
    textArea.style.top = 0;
    textArea.style.left = 0;
    textArea.style.width = "2em";
    textArea.style.height = "2em";
    textArea.style.padding = 0;
    textArea.style.margin = 0;
    textArea.style.border = "none";
    textArea.style.outline = "none";
    textArea.style.boxShadow = "none";
    textArea.style.background = "transparent";
    textArea.style.color = "transparent";

    textArea.value = content;

    const button = event.currentTarget;

    button.appendChild(textArea);

    textArea.select();

    try {
      document.execCommand("copy");

      button.setAttribute("show", "true");
      setTimeout(() => button.removeAttribute("show"), 1000);

    } catch (e) {
      //window.console.debug("could not run execCommand copy");
    }

    button.removeChild(textArea);
  };

  let title = "Send to clipboard";
  if (props.title) {
    title = props.title;
  }

  let confirmationText = "sent to clipoard";
  if (props.confirmationText) {
    confirmationText = props.confirmationText;
  }

  let iconClassName = "clipboard";
  if (props.icon) {
    iconClassName = props.icon;
  }

  let icon = null;
  let text = null;
  if (props.text) {
    if (props.icon) {
      icon = <FontAwesomeIcon icon={iconClassName} />;
    }
    text = <span>{props.text}</span>;
  } else {
    icon = <FontAwesomeIcon icon={iconClassName} />;
  }

  const classNames = ["kgs-copy-to-clipboard", props.className].join(" ");

  return (
    <span className={classNames}>
      <button role="link" onClick={clickHandler} title={title}>{icon}{text}</button>
      <div className="kgs-copy-confirmation">{confirmationText}</div>
    </span>
  );
};

CopyToClipboardButton.propTypes = {
  content: PropTypes.string,
  title: PropTypes.string,
  confirmationText: PropTypes.string,
  icon: PropTypes.string
};

export default CopyToClipboardButton;