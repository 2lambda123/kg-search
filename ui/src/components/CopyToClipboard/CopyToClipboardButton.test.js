/*
 * Copyright 2018 - 2021 Swiss Federal Institute of Technology Lausanne (EPFL)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This open source software code was developed in part or in whole in the
 * Human Brain Project, funded from the European Union's Horizon 2020
 * Framework Programme for Research and Innovation under
 * Specific Grant Agreements No. 720270, No. 785907, and No. 945539
 * (Human Brain Project SGA1, SGA2 and SGA3).
 *
 */

import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import CopyToClipboardButton from "./CopyToClipboardButton";

Enzyme.configure({ adapter: new Adapter() });

test('CopyToClipboardButton component renders initially', () => {
    const component = renderer.create(
        <CopyToClipboardButton className="className" text="text" icon="times" title="title" confirmationText="confirmation text" content="content" />
    );
  
    expect(component.toJSON()).toMatchSnapshot();
});

test('CopyToClipboardButton test className"', () => {
    const component = shallow(
        <CopyToClipboardButton className="className" text="text" icon="times" title="title" confirmationText="confirmation text" content="content" />
    );
    expect(component.hasClass("className"));
});
  
test('CopyToClipboardButton test icon"', () => {
    const component = shallow(
        <CopyToClipboardButton className="className" text="text" icon="times" title="title" confirmationText="confirmation text" content="content" />
    );
    expect(component.find("svg"));
});
  
test('CopyToClipboardButton test title', () => {
    const component = shallow(
        <CopyToClipboardButton className="className" text="text" icon="times" title="title" confirmationText="confirmation text" content="content" />
    );
    expect(component.find("button").prop("title")).toEqual('title');
});
  
test('CopyToClipboardButton test text', () => {
    const component = shallow(
        <CopyToClipboardButton className="className" text="text" icon="times" title="title" confirmationText="confirmation text" content="content" />
    );
    expect(component.find("button span").text()).toContain('text');
});
  
test('CopyToClipboardButton test confirmationText', () => {
    const component = shallow(
        <CopyToClipboardButton className="className" text="text" icon="times" title="title" confirmationText="confirmation text" content="content" />
    );
    expect(component.find("div").text()).toEqual('confirmation text');
});