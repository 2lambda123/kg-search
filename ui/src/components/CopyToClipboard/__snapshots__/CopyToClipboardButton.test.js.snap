// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyToClipboardButton component renders initially 1`] = `
<span
  className="kgs-copy-to-clipboard className"
>
  <button
    onClick={[Function]}
    role="link"
    title="title"
  >
    <span>
      text
    </span>
  </button>
  <div
    className="kgs-copy-confirmation"
  >
    confirmation text
  </div>
</span>
`;
